platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConverterService.java

```diff
@@ -128,6 +127,13 @@ <T> T convert(@Nullable final String value, @NonNull final Class<T> targetClass)
             return (T) value;
         }
         final ConfigConverter<T> converter = (ConfigConverter<T>) converters.get(targetClass);
+
+        if (converter == null && targetClass.isEnum()) {
+            // FUTURE WORK: once logging is added to this module, log a warning here
+            // ("No converter defined for type '" + targetClass + "'. Converting using backup enum converter.");
+            return (T) Enum.valueOf((Class<Enum>) targetClass, value);
```

Comment:
Can you add a try catch around that block and throw a meaningful exception in the catch?
try {
return (T) Enum.valueOf((Class<Enum>) targetClass, value);
} catch(Exception e) {
throw new IllegalArgumentException("Can not convert value FOO of Enum BAR by default. Please add a custom config converter.", e);
}

Reply:
added